### PR TITLE
Add Carrier Mapping

### PR DESCRIPTION
**Pull request by Mikewa33**
Add Carrier Mapping to Store then use mapping to update the shipping info

### CODE REVIEW

File: app/models/spree/mirakl_store.rb [hunk 1]
```diff
@@ -2,20 +2,39 @@ module Spree
...
+    def pull_in_store_carriers
+      mirakl_request = SpreeMirakl::Api.new(self)
+      request = mirakl_request.carriers()
+
+      raise Exception.new(Spree.t(:carrier_error, shop_id: self.shop_id)) unless request.success?
+
+      carriers = JSON.parse(request.body, {symbolize_names: true})[:carriers]
+
+      carriers.each do |carrier|
+        puts carrier.to_json
```
Comment (hf2186) [inline app/models/spree/mirakl_store.rb:33]: no puts here

File: app/models/spree/mirakl_store.rb [hunk 2]
```diff
@@ -2,20 +2,39 @@ module Spree
...
-  
+
     validates :name, :api_key, :url, :user_id, presence: true
     validates :name, :api_key, :url, uniqueness: { case_sensitive: false }
-  
+
     scope :active, -> { where(active: true) }
-  
+
+    after_create :pull_in_store_carriers
```
Comment (hf2186) [inline app/models/spree/mirakl_store.rb:13]: need a better name? get_carriers_from_mirakl?

File: app/jobs/mirakl_ship_job.rb [hunk 3]
```diff
@@ -11,13 +11,14 @@ def perform(store_id, order_id, shipment_id)
...
+    carrier = store.mirakl_store_carriers.where(label: shipping_name.downcase).first
 
-    if ['FEDEX','UPS','USPS'].include?(shipping_name.upcase)
+    if carrier.present?
       # registered carrier
-      shipping_info = { carrier_code: shipping_name, tracking_number: shipment.tracking }
+      shipping_info = { carrier_code: carrier.code, tracking_number: shipment.tracking }
     else
       # Unregistered shipping carrier
-      shipping_info = { carrier_name: shipping_name, tracking_number: shipment.tracking, carrier_url: shipment.mirakl_tracking_url }
```
Comment (hf2186) [inline app/jobs/mirakl_ship_job.rb:20]: why was this changed?

File: app/views/spree/admin/mirakl_stores/index.html.erb [hunk 4]
```diff
@@ -26,6 +26,7 @@
         <td class="presentation"><%= mirakl_store.active %></td>
         <td class="actions actions-2 text-right">
           <%= link_to_with_icon "stock", Spree.t(:refresh_inventory), admin_mirakl_store_refresh_inventory_path(mirakl_store), { class: 'admin_edit_mirakl_store btn btn-primary btn-sm', no_text: true } %>
+          <%= link_to_with_icon "road", Spree.t(:refresh_carriers), admin_mirakl_store_refresh_carriers_path(mirakl_store), { class: 'admin_edit_mirakl_store btn btn-primary btn-sm', no_text: true } %>
```
Comment (hf2186) [inline app/views/spree/admin/mirakl_stores/index.html.erb:29]: use package as icon?

Comment (Mikewa33) [reply app/views/spree/admin/mirakl_stores/index.html.erb]: used shipment. Couldnt find a package icon

File: app/jobs/mirakl_ship_job.rb (hunk 3 above)
Comment (Mikewa33) [reply app/jobs/mirakl_ship_job.rb:20]: Tested this yesterday to see if it is needed. Including carrier_name if it is an unknown carrier causes a 400
